### PR TITLE
[feature](merge-cloud) Add create cloud table

### PR DESCRIPTION
**Pull request by yujun777**
Co-authored-by: Luwei <[814383175@qq.com](mailto:814383175@qq.com)>
Co-authored-by: deardeng <[565620795@qq.com](mailto:565620795@qq.com)>
Co-authored-by: Gavin Chou <[gavineaglechou@gmail.com](mailto:gavineaglechou@gmail.com)>
Co-authored-by: Lightman <[31928846+Lchangliang@users.noreply.github.com](mailto:31928846+Lchangliang@users.noreply.github.com)>
Co-authored-by: zhengyu <[freeman.zhang1992@gmail.com](mailto:freeman.zhang1992@gmail.com)>
Co-authored-by: Lei Zhang <[1091517373@qq.com](mailto:1091517373@qq.com)>
Co-authored-by: AlexYue <[yj976240184@gmail.com](mailto:yj976240184@gmail.com)>
Co-authored-by: Xiaocc <[598887962@qq.com](mailto:598887962@qq.com)>
Co-authored-by: panDing19 <[56944854+panDing19@users.noreply.github.com](mailto:56944854+panDing19@users.noreply.github.com)>
Co-authored-by: plat1ko <[platonekosama@gmail.com](mailto:platonekosama@gmail.com)>
Co-authored-by: zhangdong <[493738387@qq.com](mailto:493738387@qq.com)>
Co-authored-by: walter <[w41ter.l@gmail.com](mailto:w41ter.l@gmail.com)>

## Proposed changes

TODO: 
1. analyse cloud table property;
2. analyse ttl seconds;
3. merge code of cloud internal catalog create partition

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/common/util/PropertyAnalyzer.java [hunk 1]
```diff
@@ -101,6 +101,8 @@ public class PropertyAnalyzer {
```
Comment (luwei16) [inline fe/fe-core/src/main/java/org/apache/doris/common/util/PropertyAnalyzer.java:101]: It seems that the code to force the number of replicas to 1 is missing.

File: fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java [hunk 2]
```diff
@@ -1546,7 +1549,9 @@ public void addPartition(Database db, String tableName, AddPartitionClause addPa
                     olapTable.getTimeSeriesCompactionTimeThresholdSeconds(),
                     olapTable.getTimeSeriesCompactionEmptyRowsetsThreshold(),
                     olapTable.storeRowColumn(),
-                    binlogConfig, dataProperty.isStorageMediumSpecified(), null);
+                    binlogConfig, dataProperty.isStorageMediumSpecified(), null,
+                    olapTable.getTTLSeconds());
+            afterCreatePartitions(olapTable.getId(), partitionIds, indexIds);
```
Comment (luwei16) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java:1554]: Can beforeCreatePartitions and afterCreatePartitions be called in the createPartitionWithIndices function of CloudInternalCatalog?

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Tablet.java [hunk 3]
```diff
@@ -23,7 +23,9 @@
 import org.apache.doris.common.Config;
```
Comment (luwei16) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Tablet.java:23]: These should be implemented in the cloudTablet class？

File: fe/fe-core/src/main/java/org/apache/doris/common/util/PropertyAnalyzer.java [hunk 4]
```diff
@@ -101,6 +101,8 @@ public class PropertyAnalyzer {
 
     public static final String PROPERTIES_INMEMORY = "in_memory";
```
Comment (luwei16) [inline fe/fe-core/src/main/java/org/apache/doris/common/util/PropertyAnalyzer.java:103]: It needs to be judged that the light schema change property cannot be false

File: fe/fe-core/src/main/java/org/apache/doris/common/util/PropertyAnalyzer.java (hunk 1 above)
Comment (yujun777) [reply fe/fe-core/src/main/java/org/apache/doris/common/util/PropertyAnalyzer.java:101]: 还没改，原本想TODO 的下一个pr里把cloud属性解析那里一起改了

File: fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java (hunk 2 above)
Comment (yujun777) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java:1554]: 代码里有创建多个分区的， 先在创建它们之前把所有partition先prepare， 在建完partition之后，在对所有分区再commit一次。如果要改，需要把InternalCatalog里重构一下代码，把多个分区的代码在扣出成一个函数

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run buildall

**Review by luwei16 — APPROVED** [review]

**Review by luwei16 — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/cloud/datasource/CloudInternalCatalog.java [hunk 5]
```diff
@@ -17,13 +17,495 @@
 
 package org.apache.doris.cloud.datasource;
```
[the remark this answers is not in the document]
Comment (yujun777) [reply fe/fe-core/src/main/java/org/apache/doris/cloud/datasource/CloudInternalCatalog.java:19]: ok， 后面的小pr再挨个修

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**